### PR TITLE
feat: message banner

### PR DESCRIPTION
**Pull request by nunogois**
Related to: https://linear.app/unleash/issue/2-511/exploration-build-data-for-the-possibility-of-showing-an-alert-to-the
Namely: https://unleash-internal.slack.com/archives/C046LV85N3C/p1671443897386729

The idea is to have a general message banner that can be controlled through a feature flag in Unleash to display announcements, warnings, informations, etc.

Currently using mock feature flags, but the idea is to bind this to a feature flag we can manage in our Unleash instance, and use its payload to provide information to end users whenever we want.

### CODE REVIEW

**Review by gastonfournier — COMMENTED** [review]
Is this related with Announcer? I see we might be solving the same problem twice. I also noticed the DX team has the Maintenance banner so I just wanted to double check we're not reinventing the wheel

File: frontend/src/component/common/MessageBanner/MessageBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            <StyledLink href={link} target="_blank" onClick={trackEvent}>
+                {linkText}
+            </StyledLink>
+        );
+    else
+        return (
+            <StyledLink
+                onClick={() => {
+                    trackEvent();
+                    navigate('/admin/network');
```
[the remark this answers is not in the document]
Comment (nunogois) [reply frontend/src/component/common/MessageBanner/MessageBanner.tsx]: Absolutely correct, nice catch 👍

**Comment by nunogois** [comment]
> Is this related with Announcer? I see we might be solving the same problem twice. I also noticed the DX team has the Maintenance banner so I just wanted to double check we're not reinventing the wheel

`Announcer` is related to accessibility, as you can see here:
https://github.com/Unleash/unleash/blob/e533b44c5bc08e90e116853df2f1b1c65a9bd37e/frontend/src/component/common/Announcer/AnnouncerContext/AnnouncerContext.tsx#L11-L15

The maintenance banner can eventually overlap a bit in terms of UI, but they are still fundamentally different things, unless we want to announce maintenance mode through a feature flag.

**Comment by nunogois** [comment]
Since this will always be behind a feature flag I guess it doesn't hurt to merge it for now.

File: frontend/src/component/common/MessageBanner/MessageBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    const navigate = useNavigate();
+    const tracker = usePlausibleTracker();
+    const external = link.startsWith('http');
+
+    const trackEvent = () => {
+        if (!plausibleEvent) return;
+        tracker.trackEvent('message_banner', {
+            props: { event: plausibleEvent },
+        });
```
Comment (gastonfournier) [inline frontend/src/component/common/MessageBanner/MessageBanner.tsx:119]: I like the usage of `plausibleEvent` property allowing us to track different banners independently

**Review by gastonfournier — APPROVED** [review]
I like it so far! And it's behind a flag :+1:

**Pull request merged**